client/components/ProductsPage.js

```diff
@@ -3,43 +3,59 @@ import { Banner, Layout, Card } from "@shopify/polaris";
 import React from "react";
 import { ProductsList } from "./ProductsList";
 import { Loading } from "@shopify/app-bridge-react";
+import ApplyRandomPrices from "./ApplyRandomPrices";
 
-const PRODUCTS_QUERY = gql`
-  {
-    products(first: 10) {
-      edges {
-        cursor
-        node {
-          id
-          title
-          onlineStoreUrl
+// GraphQL query to retrieve products by IDs.
+// The price field belongs to the variants object because
+// variations of a product can have different prices.
+const GET_PRODUCTS_BY_ID = gql`
+  query getProducts($ids: [ID!]!) {
+    nodes(ids: $ids) {
+      ... on Product {
+        title
+        handle
+        descriptionHtml
+        id
+        images(first: 1) {
+          edges {
+            node {
+              originalSrc
+              altText
+            }
+          }
+        }
+        variants(first: 1) {
+          edges {
+            node {
+              price
+              id
+            }
+          }
         }
       }
     }
   }
 `;
 
-function ProductsPage() {
-  const { loading, error, data } = useQuery(PRODUCTS_QUERY);
-
+function ProductsPage(props) {
+  const { productIds } = props;
```

Comment:
You could just expand the args as we get them here:

```suggestion
function ProductsPage({ productIds }) {
```

Reply:
riiight i forgot about this syntax